### ISSUE TITLE
Malformed Range header causes ETagRequestMixin.range to fail

### ISSUE THREAD

**Issue by magne4000** [pallets/werkzeug#974]
If `Range` header value is `bytes=-`, we have the following stacktrace on call to .range member of ETagRequestMixin

``` python
Traceback (most recent call last):
  File "C:\Users\1charlej\workspace\flask\flask\app.py", line 1973, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\1charlej\workspace\flask\flask\app.py", line 1626, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\1charlej\workspace\flask\flask\app.py", line 1529, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\1charlej\workspace\flask\flask\_compat.py", line 33, in reraise
    raise value
  File "C:\Users\1charlej\workspace\flask\flask\app.py", line 1624, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\1charlej\workspace\flask\flask\app.py", line 1610, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\1charlej\workspace\flask\tests\test_helpers.py", line 370, in index
    return flask.send_file('static/index.html', conditional=True)
  File "C:\Users\1charlej\workspace\flask\flask\helpers.py", line 622, in send_file
    if request.range is not None:
  File "C:\Users\1charlej\Documents\Tools\Python35\lib\site-packages\werkzeug\local.py", line 338, in __getattr__
    return getattr(self._get_current_object(), name)
  File "C:\Users\1charlej\Documents\Tools\Python35\lib\site-packages\werkzeug\utils.py", line 71, in __get__
    value = self.func(obj)
  File "C:\Users\1charlej\Documents\Tools\Python35\lib\site-packages\werkzeug\wrappers.py", line 1327, in range
    return parse_range_header(self.environ.get('HTTP_RANGE'))
  File "C:\Users\1charlej\Documents\Tools\Python35\lib\site-packages\werkzeug\http.py", line 532, in parse_range_header
    begin = int(item)
ValueError: invalid literal for int() with base 10: '-'
```

**Comment by magne4000** [comment]
Also happens with anything but digit and dash, e.g. `bytes=a-b` will also fail.

**Comment by davidism** [comment]
Closed by #977
